src/nwb_conversion_tools/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py

```diff
@@ -90,18 +108,33 @@ def get_source_schema(cls):
         source_schema["properties"]["file_path"]["description"] = "Path to SpikeGLX file."
         return source_schema
 
-    def __init__(self, file_path: FilePathType, stub_test: Optional[bool] = False):
-        super().__init__(file_path=str(file_path))
-        if stub_test:
-            self.subset_channels = [0, 1]
+    def __init__(
+        self,
+        file_path: FilePathType,
+        stub_test: Optional[bool] = False,
+        spikeextractors_backend: Optional[bool] = False,
+    ):
+
+        self.stub_test = stub_test
+        self.stream_id = fetch_stream_id_for_spikelgx_file(file_path)
+
+        if spikeextractors_backend:
+            self.RX = se.SpikeGLXRecordingExtractor
+            self.RX(file_path=file_path)
+            super().__init__(file_path=str(file_path))
+            _assert_single_shank_for_spike_extractors(self.recording_extractor)
+            self.meta = _fetch_meta_dic_for_spikextractors_spikelgx_object(self.recording_extractor)
+            self.recording_extractor = OldToNewRecording(oldapi_recording_extractor=self.recording_extractor)
+
+        else:
+            file_path = Path(file_path)
+            folder_path = file_path.parent
+            super().__init__(folder_path=folder_path, stream_id=self.stream_id)
+            self.source_data["file_path"] = str(file_path)
+            self.meta = self.recording_extractor.neo_reader.signals_info_dict[(0, self.stream_id)]["meta"]
```

Comment:
It looks like there is some duplicated code here. Is there a way of reducing it? What about a mixin class?

Reply:
For now we'll use a common auxiliary function to reduce code duplication.

Long term plan will indeed be to use the more sophisticated mixin, but that has potential for unexpected behavior if not very thoroughly tested; since this PR already does so much we might as well just leave it for a follow-up PR dedicated just to that.

Reply:
I'd be fine with an aux function, but I'd rather not add duplicate code